src/network/tarcap/packets_handlers/status_packet_handler.cpp

```diff
@@ -41,7 +40,11 @@ void StatusPacketHandler::process(const dev::RLP& packet_rlp, const PacketData&
       if (!pending_peer) {
         LOG(log_er_) << "Peer " << packet_data.from_node_id_.abridged()
                      << " missing in both peers and pending peers map - will be disconnected.";
-        host->disconnect(packet_data.from_node_id_, p2p::UserReason);
+        if (auto host_shr = host.lock(); host_shr) {
+          host_shr->disconnect(packet_data.from_node_id_, p2p::UserReason);
+        } else {
+          LOG(log_er_) << "Invalid host in StatusPacketHandler";
```

Comment:
every packet handler has its own channel so there is not really any need to specify "StatusPacketHandler".